DFE-Digital/Dfe.Academies.External
# 104117 wire up get application endpoint

As per the below ticket, wire up the GetApplication by Id method up to the API which has caused underlying enum values to change to get de-serialization working. Also, ConversionApplication object has changes to bring into line with API
https://dfe-gov-uk.visualstudio.com.mcas.ms/Academies-and-Free-Schools-SIP/_workitems/edit/104117?McasTsid=26110

## Type of change

## Expected behaviour
When going into an application via the application list page the API will be hit to grab the selected application.

Hence, only application with id=1 will now work within the UI

## Steps to reproduce issue (if relevant)
n/a

## Steps to test this PR
1. Run Unit tests - to ensure enum amendments / object amendments haven't broken anything - run auto by CI/CD
Go to https://localhost:44350/your-applications, after logging in. Click on only entry in 'pas applications' (id=1). Which will take you to:-
https://localhost:44350/application-overview?appId=1

This will now hit the academisation API and shouldn't error. Unless, id NOT = 1 then API will return a 404 error

## Prerequisites
n/a
